idea/data-storage/src/common/helpers/query-builder.ts

```diff
@@ -50,15 +50,27 @@ export function constructQueryBuilder<E extends ObjectLiteral = ObjectLiteral, K
   { fromDate, toDate }: IDates,
   offset: number,
   limit: number,
-  join?: string[],
+  join?: (string | { columns: string[]; table: string })[],
   orderBy?: { column: string; sort: 'DESC' | 'ASC' } | { column: string; sort: 'DESC' | 'ASC' }[],
 ): SelectQueryBuilder<E> {
   const alias = 't';
   const builder = repo.createQueryBuilder(alias);
 
   if (join) {
     for (const prop of join) {
-      builder.leftJoinAndSelect(`${alias}.${prop}`, prop);
+      if(typeof prop === 'object') {
+        const selectColumns = [];
+        const { columns, table } = prop;
+        builder.innerJoin(`${alias}.${table}`, table);
```

Comment:
Are you sure about inner join?

Reply:
If instead of `innerJoin` use `leftJoinAndSelect` will get all fields `meta` entity.